src/language/term/Typ.re

```diff
@@ -387,6 +412,135 @@ let unroll = (ty: t): t =>
    Other types may be equivalent but this will not detect so if they are not normalized. */
 let equal = (t1: t, t2: t): bool => fast_equal(t1, t2);
 
+// TODO Have this take a list of tys and a label and do this matching at the callsite.
+let rec project_type = (ty, label): option(t) => {
+  switch (term_of(ty), term_of(label)) {
+  | (Parens(t), _) => project_type(t, label)
+  | (Prod(tys), Label(l)) =>
+    switch (LabeledTuple.find_label(match_tup_label, tys, l)) {
+    | Some({term: TupLabel(_, ty), _}) =>
+      // LabeledTuple.find_label should return the inner ty
+      Some(ty)
+    | _ => Some(Unknown(Internal) |> temp) // TODO Consider whether we want to be doing this. We want this to be treated like a hole but it would be better to do this via a more direct error reovery mechanism
```

Comment:
There's all these cases in here that we're doing error recovery to a hole. We need to figure out a way to do this once in statics. I tried adding i to the `term` in the info map but that isn't used downstream because it also needs to be placed in the context.